### PR TITLE
Selectively hide hint indicator if not hints available.

### LINKED ISSUE #4001: For Mac OS, some Hint has no content but doesn't show system status
When there is no hint for some exercises, the users are not aware of that and will miss click and waste hints. Maybe better to show something like "No hint available" or disable the button.

### PR DESCRIPTION
**Pull request by rtibbles**
Fixes #4001

### CODE REVIEW

File: kalite/distributed/static/js/distributed/exercises/views.js [hunk 1]
```diff
@@ -712,6 +721,8 @@ var ExercisePracticeView = ExerciseWrapperBaseView.extend({
         });
 
         this.listenTo(this.exercise_view, "check_answer", this.check_answer);
+
+        this.listenTo(this.exercise_view, "hint_available", this.toggle_hint_view);
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/exercises/views.js]: `toggle_hint_view` takes an argument, but my understanding is that this form will not call toggle_hint_view with an argument. Is that correct? Can you use [_.partial](http://underscorejs.org/#partial) here?

Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/exercises/views.js]: Disregard, I see that it's handled by [Events.trigger](http://backbonejs.org/#Events-trigger).

**Comment by MCGallaspy** [comment]
I am trying to locate an exercise without hints, in order to test this out -- quite tricky, any way to search through exercises/assessment items systematically?

**Comment by rtibbles** [comment]
Would have to load up all the assessment items, parse them all, and then check to see if the hints property is an empty list or not.

I assume one must exist for the bug report to exist, but I could not find one either.

**Comment by MCGallaspy** [comment]
> Would have to load up all the assessment items, parse them all, and then check to see if the hints property is an empty list or not.

My conclusion too. Wrote a quick script:
```python
for ai in AssessmentItem.objects.all():
    item_data = json.loads(ai.item_data)
    if not len(item_data.get("hints", [])):
        print ai.id

**Comment by MCGallaspy** [comment]
Some of the assessment items for this content item have no hints, for example: `/learn/khan/computing/computer-science/algorithms/graph-representation/quiz-representing-graphs/`

Seems to work great. Merging.

**Pull request merged**